### PR TITLE
Exceptions: clean code examples with usage of custom exceptions

### PR DESCRIPTION
**Pull request by Pragatheesh**

### CODE REVIEW

File: 3-exceptions/2-after.php [hunk 1]
```diff
@@ -0,0 +1,115 @@
...
+final class PrintFailedException extends \Exception
+{
+}
+
+final class GetReportHandler
+{
+    // ...
+
+    /**
+     * @param int|null $reportId
```
Comment (eclayaz) [inline 3-exceptions/2-after.php:12]: null should be removed right

File: 3-exceptions/2-after.php [hunk 2]
```diff
@@ -0,0 +1,115 @@
...
+     * @throws InvalidArgumentException
+     */
+    public function get($reportId)
+    {
+        if ($reportId === null) {
+            throw new InvalidArgumentException("Repord id can never be empty, and needs to have a value");
+        }
+
+        /*
+         * Try-catch only for the scope and not for the full code
```
Comment (eclayaz) [inline 3-exceptions/2-after.php:24]: why not rest of the code inside the try block ?

File: 3-exceptions/2-after.php (hunk 1 above)
Comment (Pragatheesh) [reply 3-exceptions/2-after.php:12]: The caller still can call with null, but the function checks for null argument and throws exception.

But the more correct way is that we have a query object, and pass that query object to this method, and that query object would have an assert to check the report id is null. But as of this point CQRS is not yet there

File: 3-exceptions/2-after.php (hunk 2 above)
Comment (Pragatheesh) [reply 3-exceptions/2-after.php:24]: That code potentially do not throw a not found exception

**Review by eclayaz — APPROVED** [review]

**Pull request merged**